modules/videoio/misc/build_plugins.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+set -e
+
+if [ -z $1 ] ; then
+    echo "<script> <destination directory>"
+    exit 1
+fi
+
+DIR=$(readlink -f $(dirname $0))
+OCV=$(readlink -f "${DIR}/../../..")
+DST=$(readlink -f $1)
+CFG=$2
+
+do_build()
+{
+TAG=$1
+D=$2
+F=$3
+shift 3
+docker build \
+    --build-arg http_proxy \
+    --build-arg https_proxy \
+    $@ \
+    -t $TAG \
+    -f "${D}/${F}" \
+    "${D}"
+}
+
+do_run()
+{
+TAG=$1
+shift 1
+docker run \
+    -it \
+    --rm \
+    -v "${OCV}":/opencv:ro \
+    -v "${DST}":/dst \
+    -e CFG=$CFG \
+    --user $(id -u):$(id -g) \
+    $TAG \
+    $@
+}
+
+build_gstreamer()
+{
+TAG=opencv_gstreamer_builder
+do_build $TAG "${DIR}/plugin_gstreamer" Dockerfile
+do_run $TAG /opencv/modules/videoio/misc/plugin_gstreamer/build.sh /dst $CFG
+}
+
+build_ffmpeg_ubuntu()
+{
+VER=$1
+TAG=opencv_ffmpeg_ubuntu_builder:${VER}
+do_build $TAG "${DIR}/plugin_ffmpeg" Dockerfile-ubuntu --build-arg VER=${VER}
+do_run $TAG /opencv/modules/videoio/misc/plugin_ffmpeg/build-ubuntu.sh /dst ${VER} ${CFG}
+}
+
+build_ffmpeg()
+{
+VER=$1
+TAG=opencv_ffmpeg_builder:${VER}
+ARCHIVE="${DIR}/plugin_ffmpeg/ffmpeg-${VER}.tar.xz"
+if [ ! -f "${ARCHIVE}" ] ; then
+    wget https://www.ffmpeg.org/releases/ffmpeg-${VER}.tar.xz -O "${ARCHIVE}"
+fi
+do_build $TAG "${DIR}/plugin_ffmpeg" Dockerfile-ffmpeg --build-arg VER=${VER}
+do_run $TAG /opencv/modules/videoio/misc/plugin_ffmpeg/build-standalone.sh /dst ${VER} ${CFG}
+}
+
+echo "OpenCV: ${OCV}"
+echo "Destination: ${DST}"
+
+build_gstreamer
+build_ffmpeg_ubuntu 18.04
+build_ffmpeg_ubuntu 16.04
+build_ffmpeg 4.1
+build_ffmpeg 3.4.5
```

Comment:
Something is wrong with size of `libopencv_videoio_ffmpeg_static_3.4.5.so`:
```
-rwxr-xr-x 1 alalek alalek   130664 Feb  4 15:18 libopencv_videoio_ffmpeg_shared_2.8.15.so
-rwxr-xr-x 1 alalek alalek   130664 Feb  4 15:14 libopencv_videoio_ffmpeg_shared_3.4.5.so
-rwxr-xr-x 1 alalek alalek   130664 Feb  4 15:10 libopencv_videoio_ffmpeg_shared_4.1.so
-rwxr-xr-x 1 alalek alalek 77372528 Feb  4 15:18 libopencv_videoio_ffmpeg_static_2.8.15.so
-rwxr-xr-x 1 alalek alalek   126368 Feb  4 15:14 libopencv_videoio_ffmpeg_static_3.4.5.so
-rwxr-xr-x 1 alalek alalek 87701608 Feb  4 15:10 libopencv_videoio_ffmpeg_static_4.1.so
-rwxr-xr-x 1 alalek alalek   131360 Feb  4 15:04 libopencv_videoio_ffmpeg_ubuntu_16.04.so
-rwxr-xr-x 1 alalek alalek   130664 Feb  4 15:01 libopencv_videoio_ffmpeg_ubuntu_18.04.so
-rwxr-xr-x 1 alalek alalek   107864 Feb  4 14:59 libopencv_videoio_gstreamer.so
```

Reply:
Fixed